netty/src/test/java/io/grpc/netty/NettyClientTransportTest.java

```diff
@@ -284,8 +284,8 @@ public void maxHeaderListSizeShouldBeEnforcedOnServer()  throws Exception {
           + " size limit!");
     } catch (Exception e) {
       Throwable rootCause = getRootCause(e);
-      assertTrue(rootCause.getMessage(),
-          rootCause.getMessage().contains("Header size exceeded max allowed size (1)"));
+      assertTrue(rootCause instanceof StatusException);
```

Comment:
This test seems to have been broken for a while ... revealed by an update to Netty's header encoding/decoding netty/netty@540c26bb56067523e3a76e . 

On the local endpoint we set `SETTINGS_MAX_HEADER_LIST_SIZE` on the decoder, but don't send it as part of the initial `SETTINGS` to the remote endpoint. Which according to the specification is fine, as 

> For any given request, a lower limit than what is advertised MAY be enforced. The initial value of this setting is unlimited.

In this test, we (the server) set the limit on the decoder to 1 byte. Since we don't advertise the limit via `SETTINGS`, the client assumes it's unlimited and sends us a HEADERS frame. Thus an exception is triggered (on the server), and we respond with a `RST_STREAM` with code `PROTOCOL_ERROR`. All fair and square, but we don't get a detailed error message...

We can change that by including `SETTINGS_MAX_HEADER_LIST_SIZE ` in the initial `SETTINGS` frame. This would lead to the local encoder enforcing the remote decode limit [1]. However, we can't currently do that cause it seems like a bug in Netty that the encoder throws a connection error and not a stream error, bringing down the whole connection, while only the stream should fail. I ll open a issue/PR in Netty.

[1] https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Encoder.java#L126